lib/jsdom/living/structured-clone/structuredDeserializeWithTransfer.js

```diff
@@ -0,0 +1,448 @@
+"use strict";
+
+const DOMException = require("domexception/webidl2js-wrapper");
+
+function structuredDeserializeWithTransfer(serializeWithTransferResult, targetRealm) {
+  /**
+   * Let memory be an empty map.
+   * Let transferredValues be a new empty List.
+   */
+
+  const memory = {};
```

Comment:
Use `Object.create(null)` to avoid any weirdnesses if the name is something like `"__proto__"`.